### PR TITLE
Adding simple unit test presubmit and unit test periodic

### PR DESCRIPTION
**Pull request by kfswain**

### CODE REVIEW

File: config/jobs/kubernetes-sigs/gateway-api-inference-extension/inference-extension-periodics-main.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  - interval: 12h
+    name: periodic-gateway-api-inference-extension-test-unit-main
+    cluster: eks-prow-build-cluster
+    extra_refs:
+      - org: kubernetes-sigs
+        repo: gateway-api-inference-extension
+        base_ref: main
+        path_alias: sigs.k8s.io/gateway-api-inference-extension
+    annotations:
+      testgrid-dashboards: sig-apps
```
Comment (kannon92) [inline config/jobs/kubernetes-sigs/gateway-api-inference-extension/inference-extension-periodics-main.yaml:11]: Is this sig-networking or sig-apps? This should be the owning sig for the repo.

Comment (kfswain) [reply config/jobs/kubernetes-sigs/gateway-api-inference-extension/inference-extension-periodics-main.yaml]: sig net. Thanks for this! I moved it under gateway-api's dash.

**Comment by kannon92** [comment]
/lgtm
/assign @dims @cblecker 

This should get an OWNER File so the team can approve these without top-level approvers. Still then we could use your powers!

**Comment by ahg-g** [comment]
@dims @BenTheElder can you please help with this one?

**Pull request merged**